crates/oxc_linter/src/rule.rs

```diff
@@ -15,6 +15,8 @@ pub trait Rule: Sized + Default + Debug {
 pub trait RuleMeta {
     const NAME: &'static str;
 
+    const CATEGORY: &'static str;
```

Comment:
let's make this into an enum. The macro will also need to throw some error messages if it's not `correctness` or `nursery`.

Reply:
I tried 😅 https://github.com/Boshen/oxc/pull/131/commits/9cc1108db1076648930b2ce870a7637f432ad7aa